mobilecoind-json/src/main.rs

```diff
@@ -0,0 +1,317 @@
+#![feature(proc_macro_hygiene, decl_macro)]
+
+use grpcio::{ChannelBuilder, ChannelCredentialsBuilder};
+use protobuf::RepeatedField;
+
+use mc_api::external::KeyImage;
+use mc_common::logger::{create_app_logger, log, o};
+use mc_mobilecoind_api::mobilecoind_api_grpc::MobilecoindApiClient;
+use rocket::{get, post, routes};
+use rocket_contrib::json::Json;
+use serde_derive::{Deserialize, Serialize};
+use std::sync::Arc;
+use structopt::StructOpt;
+
+/// Command line config, set with defaults that will work with
+/// a standard mobilecoind instance
+#[derive(Clone, Debug, StructOpt)]
+#[structopt(
+    name = "mobilecoind-rest-gateway",
+    about = "A REST frontend for mobilecoind"
+)]
+pub struct Config {
+    /// Host to listen on.
+    #[structopt(long, default_value = "127.0.0.1")]
+    pub listen_host: String,
+
+    /// Port to start webserver on.
+    #[structopt(long, default_value = "9090")]
+    pub listen_port: u16,
+
+    /// MobileCoinD URI.
+    #[structopt(long, default_value = "127.0.0.1:4444")]
+    pub mobilecoind_host: String,
+
+    /// SSL
+    #[structopt(long)]
+    pub use_ssl: bool,
+}
+
+/// Connection to the mobilecoind client
+struct State {
+    pub mobilecoind_api_client: MobilecoindApiClient,
+}
+
+#[derive(Serialize, Default)]
+struct JsonEntropyResponse {
+    entropy: String,
+}
+
+/// Requests a new root entropy from mobilecoind
+#[get("/entropy")]
+fn entropy(state: rocket::State<State>) -> Result<Json<JsonEntropyResponse>, String> {
+    let resp = state
+        .mobilecoind_api_client
+        .generate_entropy(&mc_mobilecoind_api::Empty::new())
+        .map_err(|err| format!("Failed getting entropy: {}", err))?;
+    Ok(Json(JsonEntropyResponse {
+        entropy: hex::encode(resp.entropy),
+    }))
+}
+
+#[derive(Deserialize, Default)]
+struct JsonMonitorRequest {
+    key: String,
+    start: u64,
+    number: u64,
+}
+
+#[derive(Serialize, Default)]
+struct JsonMonitorResponse {
+    monitor_id: String,
+}
+
+/// Creates a monitor. Data for the key and range is POSTed using the struct above.
+#[post("/create-monitor", format = "json", data = "<monitor>")]
```

Comment:
Standard REST would be a POST to `/monitors` to create a new object that lives under `/monitors/...`